src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -213,6 +213,10 @@ library BeaconChainProofs {
         * Therefore, the index of the balance of a validator is validatorIndex/4
         */
         uint256 balanceIndex = uint256(validatorIndex/4);
+        /**
+        * Note: Merkleization of the balance root tree uses MerkleizeWithMixin, i.e., the length of the array is hashed with the root of 
+        * the array.  Thus we shift the BALANCE_INDEX over by BALANCE_TREE_HEIGHT + 1 and not just BALANCE_TREE_HEIGHT.
+        */
```

Comment:
I'm trying to actually fully understand this.
Looks like you're referring to this code https://github.com/ferranbt/fastssz/blob/v0.1.3/hasher.go#L269-L288 
But the spec itself is described here https://github.com/ethereum/consensus-specs/blob/28e3c807e29f0125f6871fbc06ec3465597b8175/ssz/simple-serialize.md?plain=1#L236
I can reference your proof-generation code here https://github.com/Layr-Labs/apg/blob/master/pkg/eigenlayer/proofs.go#L119-L160
which refers to the `beaconstate_ssz.go` file (seems like here https://github.com/attestantio/go-eth2-client/blob/v0.18.3/spec/deneb/beaconstate_ssz.go#L763-L776 ?)
So is the crux of the idea here that the apparent size of the container holding balances ends up being increased by 1 field because of this "mixin" step?

Reply:
yes, that's right.  Lets take a simple example:

Lets say I have array X with 8 elements.  When I merkleize it, it will have log 8 = 3 levels.  MerkleizeWithMixin takes root(X) and then hashes it again:  final_root = hash(len(X), root(X)).  So this adds an additional level to the tree.

Reply:
The "mixin" refers to mixing in the length of te array into the merkelization

Reply:
Gotcha, that makes sense. I think we can continue to improve documentation -- including clarifying this -- but I understand it's difficult to explain the complexity / nuance of these things.